### PR TITLE
[spi] add parameter to reset gpio device for spi

### PR DESCRIPTION
**Pull request by Irving-cl**
This PR is related to #4431 
Currently some GPIO is exported in some scripts to do the hardware reset. However, to use SPI, these GPIOs cannot be exported. So a parameter is added to the command to do the reset so that GPIOs don't need to be exported.

### CODE REVIEW

**Comment by Irving-cl** [comment]
@bukepo @zhanglongxia

File: src/posix/platform/openthread-system.h [hunk 1]
```diff
@@ -114,6 +114,7 @@ typedef struct otPlatformConfig
     bool        mResetRadio;            ///< Whether to reset RCP when initializing.
     bool        mRestoreDatasetFromNcp; ///< Whether to retrieve dataset from NCP and save to file.
 
+    bool     mSpiGpioReset;       ///< Whether to reset GPIO device only.
```
Comment (zhanglongxia) [inline src/posix/platform/openthread-system.h]: nit: ///< Whether to use Reset Pin to reset RCP chip only.

File: src/posix/platform/spi_interface.cpp [hunk 2]
```diff
@@ -107,6 +107,12 @@ otError SpiInterface::Init(const otPlatformConfig &aPlatformConfig)
     // Reset RCP chip.
     TrigerReset();
 
+    // Exit if only to reset GPIO device.
```
Comment (zhanglongxia) [inline src/posix/platform/spi_interface.cpp]: nit: Exit if only to reset RCP chip.

**Comment by Irving-cl** [comment]
Discussed with @xyk, this seems to be unnecessary.

**Pull request closed without merging**